### PR TITLE
Fixes disabled torrent viewer torrent download

### LINKED ISSUE #10767: Disabling Torrent Viewer doesn't disable Torrent Handling
## Test plan
1. Visit about:preferences#extensions and disable `Torrent Viewer` (if not already disabled)
2. Restart if prompted
3. Verify extension is disabled
4. Visit https://www.qubes-os.org/downloads/
5. Click `Torrent` button
6. You should be prompted to save the torrent file

#### Did you search for similar issues before submitting this one?
Yes

#### Describe the issue you encountered:
Brave's native handling of .torrent files can be problematic for some sites (#10672). In an attempt to work around this issue, I sought to disable the built-in Web Torrent extension. Unfortunately, Brave still attempts to pass .torrent requests off to Web Torrent, which is unavailable after being disabled.

#### Platform (Win7, 8, 10? macOS? Linux distro?):
Desktop

#### Brave Version (revision SHA):
0.18.23 (https://github.com/brave/browser-laptop/commit/36ae2ec06efed0057ae16351108bf4d4c10a7ab4)

#### Steps to reproduce:
1. Disable Torrent Viewer in about:preferences#extensions (and restart Brave)
2. Open Brave, then attempt watch a video on bitchute.com
3. Note that Brave attempts to pass .torrent requests to extension
    
#### Actual result:
Brave attempts to handle torrent requests long after Torrent Viewer has been disabled.

#### Expected result:
Disabling Torrent Viewer causes Brave to leave .torrent requests alone.

#### Will the steps above reproduce in a fresh profile? If not what other info can be added?
Yes

#### Is this an issue in the currently released version?
Yes

#### Can this issue be consistently reproduced?
Yes

#### Screenshot if needed:

image

#### Any related issues:
#10672

### PR DESCRIPTION
**Pull request by NejcZdovc**
Resolves #10767

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request [as needed](https://github.com/brave/browser-laptop/wiki/Pull-request-process).
- [ ] Request a security/privacy review [as needed](https://github.com/brave/handbook/blob/master/development/security.md#how-to-request-a-security-review). (Ask a Brave employee to help if you cannot access this document.)

## Test Plan:

1. disable Torrent Viewer in preferences
2. go to https://webtorrent.io/free-torrents/
3. download Big Buck Bunny torrent file
4. make sure that save dialog is opened

## Reviewer Checklist:

- [ ] Request a security/privacy review [as needed](https://github.com/brave/handbook/blob/master/development/security.md#how-to-request-a-security-review) if one was not already requested.

Tests

- [ ] Adequate test coverage exists to prevent regressions
- [ ] Tests should be independent and work correctly when run individually or as a suite [ref](https://github.com/brave/browser-laptop/wiki/Code-Guidelines#test-dependencies)
- [ ] New files have MPL2 license header

### CODE REVIEW

File: js/stores/appStore.js [hunk 1]
```diff
@@ -202,6 +204,14 @@ const applyReducers = (state, action, immutableAction) => reducers.reduce(
       return newState
     }, appState)
 
+const isWebTorrentEnabled = (state) => {
```
Comment (NejcZdovc) [inline js/stores/appStore.js]: Place of this function is not perfect. Where would you put it, so that I can also add unit tests?

cc @bsclifton @petemill

Comment (petemill) [reply js/stores/appStore.js]: @NejcZdovc how about `app/common/state/extensionState.js`?

Comment (NejcZdovc) [reply js/stores/appStore.js]: I was thinking about that file but was not sure. Ok let me put it there. Thx

Comment (bsclifton) [reply js/stores/appStore.js]: that is a perfect place to put it- good call  😄

Comment (NejcZdovc) [reply js/stores/appStore.js]: moved

**Comment by bsclifton** [comment]
master 9f02082aaecd8e1f2a1e73a011caab642c3cac78
0.23.x 4ab36a39a00730e1a6cf93d12931c5134e512349
0.22.x e30b3c54661290521c01257e5363e89a2af9323d

**Pull request merged**